src/common/cluster-store.ts

```diff
@@ -372,3 +372,9 @@ export function getHostedClusterId() {
 export function getHostedCluster(): Cluster {
   return clusterStore.getById(getHostedClusterId());
 }
+
+export function reportDeadClusters() {
+  clusterStore.clustersList.filter(cluster => cluster.isDead).forEach(cluster => {
```

Comment:
More about an issue in the target branch/PR but could we actually store "dead state" in `ClusterManager` and not in cluster object itself? That way we would not need to add yet-another-attribute to a (exported) class that's also visible through the extension API.

Reply:
Played around this a bit and it's not that trivial. If we raise error from `Cluster` contructor and catch it on `ClusterStore`, cluster object itself will be `undefined` and cannot put in to cluster store items -> it will get lost when persisting cluster store next time.

Reply:
Exactly why I think this should be stored in `ClusterStore`. A `Cluster` represents a valid kubeconfig.

Reply:
I have one idea that I will try, but I agree it's more about the issue in the target PR, so I propose let's continue discussion there.